pkg/repositories/gormimpl/common.go

```diff
@@ -114,3 +142,39 @@ func applyScopedFilters(tx *gorm.DB, inlineFilters []common.InlineFilter, mapFil
 	}
 	return tx, nil
 }
+
+func getGroupByForNamedEntity(tableName string) string {
+	return fmt.Sprintf("%s.%s, %s.%s, %s.%s, %s.%s", tableName, Project, tableName, Domain, tableName, Name, namedEntityMetadataTableName, Description)
+}
+
+func getSelectForNamedEntity(tableName string) []string {
+	return []string{
+		fmt.Sprintf("%s.%s", tableName, Project),
+		fmt.Sprintf("%s.%s", tableName, Domain),
+		fmt.Sprintf("%s.%s", tableName, Name),
+		fmt.Sprintf("%s.%s", namedEntityMetadataTableName, Description),
+	}
+}
+
+func getNamedEntityFilters(resourceType core.ResourceType, project string, domain string, name string) ([]common.InlineFilter, error) {
+	entity := common.ResourceTypeToEntity[resourceType]
+
+	filters := make([]common.InlineFilter, 0)
+	projectFilter, err := common.NewSingleValueFilter(entity, common.Equal, Project, project)
```

Comment:
btw you can use https://github.com/lyft/flyteadmin/blob/master/pkg/manager/impl/util/filters.go#L146 for these

Reply:
So, I was trying to maintain package separation. It seemed to me like the idea is that Manager can reach down and use methods from Repo. But we had no instances of Repo reaching *up* to use things from Manager. I'm not sure how Go handles circular dependencies, but that would create one. I only really saw the Repo using methods from the common package.
But it also seemed overkill/unnecessary to try to move all of those things out of the Manager layer when they seem pretty clearly specific to it.